src/components/menu/menu-item-custom-control/menu-item-custom-control.directive.ts

```diff
@@ -0,0 +1,61 @@
+import { FocusableOption, FocusOrigin } from '@angular/cdk/a11y';
+import { Directive, ElementRef, Inject, InjectionToken, OnDestroy, OnInit, Optional, Renderer2 } from '@angular/core';
+import { takeUntil } from 'rxjs/operators';
+import { FocusIndicatorService } from '../../../directives/accessibility';
+import { MenuItemType } from '../menu-item/menu-item-type.enum';
+import { MenuTabbableItemDirective } from '../menu-tabbable-item/menu-tabbable-item.directive';
+import { MenuComponent } from '../menu/menu.component';
+
+export const FocusableItemToken = new InjectionToken<FocusableOption>('Focusable Option');
+
+@Directive({
+    selector: '[uxMenuItemCustomControl]',
+    host: {
+        '[class.ux-menu-item]': 'true',
+        'role': 'menuitem'
+    }
+})
+export class MenuItemCustomControlDirective extends MenuTabbableItemDirective implements FocusableOption, OnInit, OnDestroy {
+
+    /** Indicate the type of the menu item */
+    readonly type: MenuItemType = MenuItemType.Custom;
+
+    constructor(
+        readonly _menu: MenuComponent,
+        readonly _elementRef: ElementRef<HTMLElement>,
+        readonly _focusIndicatorService: FocusIndicatorService,
+        readonly _renderer: Renderer2,
+        @Inject(FocusableItemToken) @Optional() private readonly _focusableOption: FocusableOption
+    ) {
+        super(_menu, _elementRef, _focusIndicatorService, _renderer);
+    }
+
+
+    ngOnInit(): void {
+        // register this item in the MenuComponent
+        super.ngOnInit();
+
+        this._menu.opened
+            .pipe(takeUntil(this._onDestroy$))
+            .subscribe(() => {
+                this.removeInternalTabIndex();
+            });
+    }
+
+    /** Focus this item with a given origin */
+    focus(origin: FocusOrigin): void {
+        super.focus(origin);
+        this._focusableOption ? this._focusableOption.focus(origin) : this._elementRef.nativeElement.focus();
+    }
+
+    // remove any existing tab index on component instance and have it handled by this directive
+    removeInternalTabIndex(): void {
```

Comment:
Unless we override the tabindex in any of the 3 components that we pass the token to? If this directive is applied then prevent any tabindex binding?

We are just trying to prevent the user from being able to hit the tab key to move between the menu items, it should only be controlled by the arrow keys. Unless there would be a better way?

Reply:
I don't know if it would work or if there are any side effects, but could we listen for tab keydown events and preventDefault on them in the menu custom item directive instead?